beetsplug/mpdupdate.py

```diff
@@ -112,18 +75,44 @@ def __init__(self):
             if self.config[key].exists():
                 config['mpd'][key] = self.config[key].get()
 
+        self.register_listener('database_change', self.db_change)
+
+    def db_change(self, lib):
+        self.register_listener('cli_exit', self.update)
```

Comment:
Don't worry about it; let's leave it like this for now. We can switch back if it becomes an issue.

(As an aside: One argument for the admittedly less-elegant `self.database_changed = True` style is that it makes the state explicit—there's a variable that shows *inside the plugin* exactly what's going on. Otherwise, we're sort of hiding the mechanism elsewhere in the beets code.)

Maybe the right solution eventually is to add this logic to `BeetsPlugin` so that individual plugins don't have to reinvent it. That is, we'd do something like add a `exit_after_changed` event that neatly encapsulates this strategy.